Support rendering all `sharedSection` types

j0nas opened the pull request:
# Support rendering all `sharedSection` types

## Intent 

Support ~all~ most `sharedSection` types as currently defined in _/studio/schemas/sections/index.ts_.
Does not add support for `sponsorshipsSection` as that is handled separately in #66. 

## Description

Apologies for this all-over-the-place PR. I was fixing each component as I was adding/fixing Studio support, so this PR is a mix of query updates and individual component refactors to support ever-changing Studio schemas.

I've created a [Route](https://stagingadmin.structuredcontent.live/desk/route;36c8bda6-e3a3-4671-a260-cdaf3dcd2e43) containing most (if not all) of the `sharedSection` types we're expected to support [here](https://structured-content-2022-web-m1k5lppgv.sanity.build/shared-sections-test). Not every section is going to look pretty, as a) we've not gotten sketches for all, and b) I'm not even sure if we need to support every variant, as I've not seen e.g. "highlighted" versions of e.g. Venues. 

Ideally, I'd also migrate all the components to use `PortableTextComponentProps<[COMPONENT]Props>` typings and refactor _SharedSections.tsx_ to not use the `{{}... as any}` hack to appease TS, but.. time. Post-launch, I guess. I'm submitting this PR in what I feel is a somewhat disheveled state so that it can get merged and I can start working in smaller increments again.


## Testing this PR

Verify that the /shared-sections-test contents mostly map to the respective Landing Page content that's configured Studio.

oyvind-stenhaug commented on web/components/ConferenceUpdatesForm/ConferenceUpdatesForm.tsx:
```diff
@@ -1,55 +1,69 @@
+
+        <div className={styles.mainContents}>
+          <form
+            action={target}
```
Where does this `target` come from? It seems to be undefined, is that known?

oyvind-stenhaug replied:
Nit: Should we output the form (or really, this section) at all if it doesn't lead anywhere?

oyvind-stenhaug commented on web/components/TextBlock/SharedSections/SharedSections.tsx:
```diff
@@ -20,34 +23,53 @@ export const SharedSections = ({ value: { name, sections, ...rest } }) => (
-        case 'mailchimp':
-        case 'mailchimpSection':
+        case 'figure':
+          return <Figure key={section._key} value={section} />;
```
Any reason this is still using an inlined `Figure`? Can it use the proper Figure component instead (currently in web/components/TextBlock/Figure/`?

j0nas replied:
Great catch. 3077d0d.

oyvind-stenhaug reviewed (changes requested):
I can't really review or test this properly, but I left a couple of comments.

Also, is `ticketsSection` missing as a `case` in SharedSections.tsx, or is that related to a separate PR/intentionally not included in "most `sharedSection` types"?

oyvind-stenhaug reviewed (approved):
"Rubber-stamped"

j0nas commented:
@oyvind-stenhaug 

> Also, is ticketsSection missing as a case in SharedSections.tsx, or is that related to a separate PR/intentionally not included in "most sharedSection types"?

Good catch. I was going blind working with this PR. I added `<Tickets  />` support in 9e4b31b, and added both "Highlighted" and "All" Tickets shared section to the test route in Sanity. Can be seen at the bottom here: https://structured-content-2022-web-c8tu4jgai.sanity.build/shared-sections-test

oyvind-stenhaug commented:
Thanks, now the tickets table actually shows up on /registration-info again too.
